lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -266,16 +272,25 @@ export function parseCatalog(
         skipReason,
       };
     } else {
-      dependency = {
-        ...dependencyBase,
-        currentValue,
-        managerData: { fileReplacePosition },
-      };
+      const groupName = isVersionPointer(version) ? version.ref : null;
+      dependency = is.nullOrUndefined(groupName)
+        ? {
+            ...dependencyBase,
+            currentValue,
+            managerData: { fileReplacePosition },
+          }
+        : {
+            ...dependencyBase,
+            currentValue,
+            managerData: { fileReplacePosition },
+            groupName,
+          };
     }
 
     extractedDeps.push(dependency);
   }
-  return extractedDeps.map((dep) =>
+  const deps = extractedDeps.map((dep) =>
     deepmerge(dep, { managerData: { packageFile } })
```

Comment:
I don't like this `deepmerge`, but let's not touch it for now.
Instead, let's also add processing from `gradle/extract.ts`, i.e. this logic:
```ts
if (
  !is.nullOrUndefined(dep.groupName) &&
  versionRefs.has(dep.groupName)
) {
  delete dep.commitMessageTopic;
}
```